From 28c80079836617ea45283e4536068f63fa04c1ac Mon Sep 17 00:00:00 2001
From: Prem Sichanugrist <s@sikachu.com>
Date: Fri, 25 Feb 2011 16:51:53 +0700
Subject: [PATCH 1/2] Make sure that we set `env["action_dispatch.show_exceptions"]`

This has been used by `show_exception` middleware even the setting wasn't get passed to `env` hash.

(This is the backported version to 3-0-stable)
---
 railties/lib/rails/application.rb               |    3 ++-
 railties/test/application/configuration_test.rb |   15 +++++++++++++++
 2 files changed, 17 insertions(+), 1 deletions(-)

diff --git a/railties/lib/rails/application.rb b/railties/lib/rails/application.rb
index 5b26333..c306d6a 100644
--- a/railties/lib/rails/application.rb
+++ b/railties/lib/rails/application.rb
@@ -171,7 +171,8 @@ module Rails
     def env_defaults
       @env_defaults ||= {
         "action_dispatch.parameter_filter" => config.filter_parameters,
-        "action_dispatch.secret_token" => config.secret_token
+        "action_dispatch.secret_token" => config.secret_token,
+        "action_dispatch.show_exceptions" => config.action_dispatch.show_exceptions
       }
     end
 
diff --git a/railties/test/application/configuration_test.rb b/railties/test/application/configuration_test.rb
index d63d25b..3e0a0dc 100644
--- a/railties/test/application/configuration_test.rb
+++ b/railties/test/application/configuration_test.rb
@@ -277,5 +277,20 @@ module ApplicationTests
       get "/"
       assert_not_equal res, last_response.body
     end
+
+    test "config.action_dispatch.show_exceptions is sent in env" do
+      make_basic_app do |app|
+        app.config.action_dispatch.show_exceptions = true
+      end
+
+      class ::OmgController < ActionController::Base
+        def index
+          render :text => env["action_dispatch.show_exceptions"]
+        end
+      end
+
+      get "/"
+      assert_equal 'true', last_response.body
+    end
   end
 end
-- 
1.7.4.1

